### PR TITLE
Add jinja filter merge_list_of_dicts

### PR DESCRIPTION
**Pull request by jkupferer**
Given a list of the form:

`[{"a": "A"}, {"b": "B"}]`

Return

`{"a": "A", "b": "B"}`

This is needed for Babylon to produce combined provision data.

### CODE REVIEW

**Review by ahsen-shah — APPROVED** [review]

**Review by fridim — APPROVED** [review]

File: test/unittest-poolboy_templating.py [hunk 1]
```diff
@@ -390,5 +390,14 @@ def test_26(self):
             recursive_process_template_strings(template, 'jinja2', template_vars), {"a": "A"}
         )
 
+    def test_27(self):
+        template = "{{ l | merge_list_of_dicts | object }}"
+        template_vars = {
+            "l": [{"a": "A", "b": "X"}, {"b": "B", "c": "C"}, {"d": "D"}]
```
Comment (fridim) [inline test/unittest-poolboy_templating.py:396]: edge cases? empty dicts, but also what's the behavior in case of non-dict values? `None`,  `false`, `[]`, etc.

Comment (jkupferer) [reply test/unittest-poolboy_templating.py:405]: Added handling for None values. Other non-dictionary values can throw an error.
